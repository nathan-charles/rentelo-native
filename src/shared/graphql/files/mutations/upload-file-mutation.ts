import gql from 'graphql-tag';
import { ReactNativeFile } from 'apollo-upload-client';

export interface UploadFileMutationData {
  createFile: {
    name: string;
    url: string;
  };
}

export interface UploadFileMutationVariables {
  upload: ReactNativeFile;
}

const UPLOAD_FILE_MUTATION = gql`
  mutation UploadFileMutation($upload: Upload!) {
    createFile(upload: $upload) {
      name
      url
    }
  }
`;

export { UPLOAD_FILE_MUTATION };
