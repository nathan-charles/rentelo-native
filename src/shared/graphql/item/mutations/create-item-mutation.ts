import gql from 'graphql-tag';

export interface Owner {
  __type: string;
  className: string;
  id: string;
}

export interface CreateItemMutationData {
  data: {
    objects: {
      createItem: {
        id: string;
      };
    };
  };
}

export interface CreateItemMutationVariables {
  fields: {
    name: string;
    description: string;
    owner: Owner;
    pricePerHour?: number;
    pricePerDay?: number;
  };
}

const CREATE_ITEM_MUTATION = gql`
  mutation CreateItemMutation($fields: ItemCreateFields) {
    objects {
      createItem(fields: $fields) {
        id
      }
    }
  }
`;

export { CREATE_ITEM_MUTATION };
