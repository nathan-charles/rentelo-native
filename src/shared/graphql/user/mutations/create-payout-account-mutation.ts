import gql from 'graphql-tag';

export interface PayoutAccountInput {
  firstName: string;
  lastName: string;
  dob: string;
  ssn: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  state: string;
  zip: string;
  phone: string;
  email: string;
  bankRoutingNumber: string;
  bankAccountNumber: string;
}

export interface CreatePayoutAccountMutationData {
  data: {
    createPayoutAccount: {
      accountId: string;
      payoutsEnabled: boolean;
    };
  };
}

export interface CreatePayoutAccountMutationVariables {
  input: PayoutAccountInput;
}

const CREATE_PAYOUT_ACCOUNT_MUTATION = gql`
  mutation CreatePayoutAccountMutation($input: PayoutAccountInput) {
    createPayoutAccount(input: $input) {
      accountId
      payoutsEnabled
    }
  }
`;

export { CREATE_PAYOUT_ACCOUNT_MUTATION };
