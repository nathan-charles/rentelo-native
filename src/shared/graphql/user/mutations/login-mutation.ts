import gql from 'graphql-tag';

export interface LoginMutationData {
  logIn: {
    sessionToken: string;
  };
}

export interface LoginMutationVariables {
  username: string;
  password: string;
}

const LOGIN_MUTATION = gql`
  mutation LoginMutation($username: String!, $password: String!) {
    logIn(fields: { username: $username, password: $password }) {
      sessionToken
    }
  }
`;

export { LOGIN_MUTATION };
