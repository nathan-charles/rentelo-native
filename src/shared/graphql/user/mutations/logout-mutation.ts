import gql from 'graphql-tag';

const LOGOUT_MUTATION = gql`
  mutation LogoutMutation {
    logOut {
      # id
      # updatedAt
      # createdAt
      # username
      sessionToken
      # ACL
    }
  }
`;

export { LOGOUT_MUTATION };
