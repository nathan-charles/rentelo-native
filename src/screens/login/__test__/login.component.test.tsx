import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import { MockedProvider, wait } from '@apollo/react-testing';

import Login from '../';
import { LOGIN_MUTATION } from '../../../shared/graphql/user/mutations/login-mutation';

const username = 'nathan.charles@me.com';
const password = 'password';
let logInMutationCalled = false;

const mocks = [
  {
    request: {
      query: LOGIN_MUTATION,
      variables: {
        username,
        password,
      },
    },
    result: () => {
      logInMutationCalled = true;

      return {
        data: {
          logIn: { sessionToken: '123' },
        },
      };
    },
  },
];

describe('<Login />', () => {
  test('renders correctly', async () => {
    const props: any = {
      navigation: { navigate: jest.fn() },
    };

    const component = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Login {...props} />
      </MockedProvider>
    );

    await wait(1);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('login mutation', async () => {
    const props: any = {
      navigation: { navigate: jest.fn() },
    };

    const component = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Login {...props} />
      </MockedProvider>
    );

    const { getByTestId } = component;

    fireEvent(getByTestId('input-email'), 'onChangeText', username);
    fireEvent(getByTestId('input-password'), 'onChangeText', password);
    fireEvent.press(getByTestId('button-login'));

    await wait(1);

    expect(logInMutationCalled).toBe(true);

    // const tree = component.toJSON();
    // expect(tree.children).toContain('Error!');
  });
});
