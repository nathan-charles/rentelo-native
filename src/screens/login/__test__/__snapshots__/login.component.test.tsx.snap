// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#c42933",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "height": 1334,
        },
        Object {
          "backgroundColor": "#F63440",
        },
      ]
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          Object {
            "flexGrow": 1,
            "justifyContent": "center",
            "padding": 15,
          },
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginBottom": 40,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontSize": 60,
                  "fontWeight": "bold",
                }
              }
            >
              Rentelo
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#FFF",
                  "fontSize": 15,
                  "fontStyle": "italic",
                  "fontWeight": "bold",
                  "marginTop": -10,
                }
              }
            >
              Save and earn cash today!
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#c42933",
              "borderRadius": 5,
              "flexDirection": "row",
              "marginTop": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 0,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#F6868D",
                    "marginLeft": 0,
                  },
                ],
                Object {
                  "fontFamily": "material",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            type="MaterialIcons"
          >
            
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            editable={true}
            keyboardAppearance="dark"
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="Email Address"
            placeholderTextColor="#F6868D"
            rejectResponderTermination={true}
            selectionColor="#FFF"
            style={
              Array [
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 17,
                  "height": 50,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                Object {
                  "color": "#FFF",
                },
              ]
            }
            testID="input-email"
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#c42933",
              "borderRadius": 5,
              "flexDirection": "row",
              "marginTop": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 0,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#F6868D",
                    "marginLeft": 0,
                  },
                ],
                Object {
                  "fontFamily": "material",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            type="MaterialIcons"
          >
            
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={true}
            keyboardAppearance="dark"
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="#F6868D"
            rejectResponderTermination={true}
            secureTextEntry={true}
            selectionColor="#FFF"
            style={
              Array [
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 17,
                  "height": 50,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                },
                Object {
                  "color": "#FFF",
                },
              ]
            }
            testID="input-password"
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
          />
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "marginTop": 15,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontSize": 14,
                "fontWeight": "500",
              }
            }
          >
            Forgot Password?
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#FFF",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginTop": 20,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
          testID="button-login"
        >
          <Text
            style={
              Object {
                "color": "#F63440",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            LOG IN
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
